### ISSUE TITLE
Problem installing sfdx-git-delta

### ISSUE THREAD

**Issue by josme12698** [scolladon/sfdx-git-delta#681]
Issue verification check:

- [x] is the current repository fully deployable at the commit SHA provided with the 'from' parameter of the command?

## What is the problem?

---

## Steps to reproduce

---

## Execution context

---

**Operating System:** …

**yarn version:** …

**node version:** node:20.0.0

**git version:** …

**sfdx version:** 7.209.6-8ba3197

**sgd plugin version:** …

**Comment by scolladon** [comment]
Hi @josme12698 !

Thanks for raising this issue and thanks for contributing in making this project better!

What is the issue ? 
What is the plugin version, the os type, the git version and the yarn version ?

**Comment by josme12698** [comment]
Hi @scolladon 
I'm so sorry, I thought I had uploaded an image with the problem. There is more information.

We are having problem in our gitlab pipeline.
We have been using this plugin without any problems. But since yesterday this step in our pipeline gitlab is failing.

generate-delta-package:
    stage: generate-delta-package
    script:
        - export CLIURL=https://developer.salesforce.com/media/salesforce-cli/sfdx-linux-amd64.tar.xz
        - mkdir sfdx
        - wget -qO- $CLIURL | tar xJ -C sfdx --strip-components 1
        - "./sfdx/install"
        - export PATH=./sfdx/$(pwd):$PATH
        - echo "install SFDX-Git-Delta plugin - https://github.com/scolladon/sfdx-git-delta"
        - sfdx update
        - echo 'y' | sfdx plugins:install sfdx-git-delta
        - git fetch origin "+refs/heads/*:refs/remotes/origin/*"
        - if [[ $COMMIT_PACKAGE == yes ]]; then
        - sfdx sgd:source:delta --to "$BRANCH_TARGET" --from $CI_COMMIT_FROM --output .
        - else
        - sfdx sgd:source:delta --to origin/"$CI_COMMIT_REF_NAME" --from $BRANCH_TARGET --repo . --output .
        - fi
        - echo "--- package.xml Generated with added and modified metadata ---"
        - cat package/package.xml
        - echo "--- destructiveChanges.xml generated with added and modified metadata ---"
        - cat destructiveChanges/destructiveChanges.xml
    artifacts:
        paths:
        - package/package.xml
        - destructiveChanges/destructiveChanges.xml
        
        Specifically in this script echo 'y' | sfdx plugins:install sfdx-git-delta when is running is failing indicated this.
install SFDX-Git-Delta plugin - https://github.com/scolladon/sfdx-git-delta
$ echo 'y' | sfdx plugins:install sfdx-git-delta
Finished digital signature check.
This plugin is not digitally signed and its authenticity cannot be verified. Continue installation y/n?: Installing plugin sfdx-git-delta... yarn add v1.22.10
Installing plugin sfdx-git-delta... info No lockfile found.Installing plugin sfdx-git-delta... [1/4] Resolving packages...Installing plugin sfdx-git-delta... [2/4] Fetching packages...Installing plugin sfdx-git-delta... info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.Installing plugin sfdx-git-delta... failed
warning sfdx-git-delta > @salesforce/command@5.3.9: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
error fs-extra@11.1.1: The engine "node" is incompatible with this module. Expected version ">=14.14". Got "12.18.3"
error Found incompatible module.
    Error: yarn add sfdx-git-delta@latest --non-interactive 
    --mutex=file:/root/.local/share/sfdx/yarn.lock 
    --preferred-cache-folder=/root/.cache/sfdx/yarn --check-files exited with 
    code 1
    
    
    Just let me know if you need any additional information.

**Comment by scolladon** [comment]
Thanks for those information.

It seems the issue comes from the node version `(The engine "node" is incompatible with this module. Expected version ">=14.14". Got "12.18.3")`.
It can come from multiple places.

- from the actual node version installed.
- from the node version coming with the cli when installed from the download url.

My suggestion would be to use node lts and to install the cli via npm (`npm install -g sfdx-cli`). 

Do you have those information (`$ uname -v ; yarn -v ; node -v ; git --version ; sfdx --version ; sfdx plugins`):
Operating System: …
yarn version: …
node version: ...
git version: …
sfdx version: ...
sgd plugin version: …

**Comment by pietroz123** [comment]
I'm having the same issue installing the plugin. Can someone help? Thanks a lot!

![image](https://github.com/scolladon/sfdx-git-delta/assets/38860782/258647af-cd1f-4cbe-bab8-af1f5027889b)

**Comment by scolladon** [comment]
Hi @pietroz123 

Sorry to ear that, could you share the exact issue you have please ?

Sometimes (I don't know why), but uninstalling sfdx and reinstalling it unblock the plugin installation.

**Comment by pietroz123** [comment]
Hi @scolladon , thanks for the quick response. The issue is the same @josme12698 had. Please see below:

```
sfdx plugins:install sfdx-git-delta
 ›   Warning: @salesforce/cli update available from 2.23.20 to 2.28.6.
? This plugin isn't signed by Salesforce. Only install the plugin if you trust its creator. Do you want to continue the installation?, Yes
Because you approved this plugin, you can avoid future installation confirmations by adding the plugin to the unsignedPluginAllowList.json file. For details, see https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_allowlist.htm.
Finished digital signature check.
Installing plugin sfdx-git-delta@latest... failed
 ›   Warning: sfdx-git-delta > @salesforce/command@5.3.9: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
    Error: /usr/local/lib/node_modules/@salesforce/cli/node_modules/yarn/bin/yarn.js add sfdx-git-delta@latest --non-interactive --mutex=file:/Users/pietrozuntini/.local/share/sf/yarn.lock --preferred-cache-folder=/Users/pietrozuntini/Library/Caches/sf/yarn --check-files exited with 
    code 1
```

**Comment by scolladon** [comment]
Ok, could you update `sfdx` cli and try again please ?
I want to make sure you are up to date with everything

**Comment by pietroz123** [comment]
Hi @scolladon !

I've updated sfdx and it still didn't work. But then I used sudo and the plugin was installed, but I still got warnings, I guess sudo just bypassed them. Here is the output:

![image](https://github.com/scolladon/sfdx-git-delta/assets/38860782/1352e11d-a004-4ea0-8f7d-baf073c81859)

Also, when executing the command to get the diff, although it worked and generated both the package and destructiveChanges folders with the correct files, I also got and warning:

```
sfdx sgd:source:delta --from origin/master --output "." 
(node:36056) Warning: Deprecated config name: apiVersion. Please use org-api-version instead.
(Use `node --trace-warnings ...` to show where the warning was created)
{
  "error": null,
  "output": ".",
  "success": true,
  "warnings": []
}
```

In conclusion, it worked but I thought I'd send you the warnings just in case. Maybe with a clean installation when I have the time the warnings will disappear.

Thanks!

**Comment by scolladon** [comment]
Thanks for the information @pietroz123 
Too bad it requires you to install the plugin with sudo, maybe the sfdx cli has been installed with sudo and that's why you also need sudo to install plugins.

Thanks for the warnings as well, we have another [issue](https://github.com/scolladon/sfdx-git-delta/issues/619) to track it
